dut/drivers.wake

```diff
@@ -0,0 +1,59 @@
+
+tuple SkeletalDriverPlan =
+  global DuhFile:             Path
+  global CompanyName:         String
+  global DeviceName:          String
+  global LocalMetalDirectory: String
+  global DUTPlan:             RocketChipDUTPlan
+
+global def makeSkeletalDriverPlan = SkeletalDriverPlan _ _ _ _ _
+
+def getDevicePlan deviceName =
+  def plans = subscribe skeletalDriverPlans
+  def samePlan aPlan =
+    if aPlan.getSkeletalDriverPlanDeviceName ==~ deviceName
+    then Some aPlan
+    else None
+  mapPartial samePlan plans
+  | head
+
+def skeletalPlanToPlan skeletalDriverPlan =
+  def objModel =
+    makeRTL (skeletalDriverPlan.getSkeletalDriverPlanDUTPlan)
+    | getDUTObjectModel
+    | prettyJSON
+
+  def name =
+    skeletalDriverPlan
+    | getSkeletalDriverPlanDUTPlan
+    | getRocketChipDUTPlanName
+
+  def omfile = write "{workspace}/build/api-generator-sifive/{name}/objectmodel.json" objModel
+  def cmd =
+    (which "python3.7"),
+    "{here}/../scripts/om2files.py".simplify,
+    omfile.getPathName,
+    skeletalDriverPlan.getSkeletalDriverPlanDuhFile.getPathName,
+    skeletalDriverPlan.getSkeletalDriverPlanCompanyName,
+    skeletalDriverPlan.getSkeletalDriverPlanDeviceName,
+    skeletalDriverPlan.getSkeletalDriverPlanLocalMetalDirectory,
+    Nil
+  def visibleFiles =
+    "{here}/../scripts/om2files.py".simplify.source,
+    skeletalDriverPlan.getSkeletalDriverPlanDuhFile,
+    omfile,
+    skeletalDriverPlan.getSkeletalDriverPlanLocalMetalDirectory.source,
+    Nil
+
+  makePlan cmd visibleFiles
+
+def makeSkeletalDrivers skeletalDriverPlan =
+  skeletalPlanToPlan skeletalDriverPlan
+  | addPythonEnv ("{here}/../scripts/om2files_env".simplify)
+  | runJob
+  | getJobStderr
+
+global def createSkeletalDriver deviceName =
```

Comment:
For all `global` functions can we please write a comment document what it does, what it takes in, and what it returns?

I'm also kind of confused by the return type here. It looks like it's an `Option (Result String Error)`, where the `Some (Pass String)` case is the standard output text. What is a caller meant to do with the return type? Shouldn't we make sure that the return type actually includes Paths to the generated driver files?

Reply:
fixed